src/controllers/users.controller.ts

```diff
@@ -60,6 +60,18 @@ class UsersController {
       next(error);
     }
   };
+
+  public manageGroups = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const data = req.body;
+      const userId = Number(req.params.id);
+      const groupData = await this.userService.manageGroups(userId, data);
+
+      res.status(200).json({ data: groupData, message: 'manageGroups' });
+    } catch (error) {
+      next(error);
+    }
+  }
 }
 
```

Comment:
1. Error handling: To enhance error handling, use a global error handling middleware. This decouples your code and gives you a single place to handle all errors in your system.
2. Business logic should be in services: UserController’s methods should contain only logic for handling requests and responses. Maintain all business logic, including interactions with Prisma, data transformation, validations, etc., inside the UserService.
3. Dependency injection: Instead of creating an instance of userService directly in the class, inject it through the constructor. This increases maintainability and testability of the code.
4. Validation: Use a validation framework such as Joi or Yup to validate request parameters and request body.
5. Types: Ensure all methods have the appropriate return types. For instance, the manageGroup function is not currently returning a Promise.
6. Security: Make sure to sanitize user input and use a middleware for authorization/authentication to ensure secure access to endpoints.

Also Don't Repeat Yourself (DRY)

You can create a higher-order function (a function that accepts function as a parameter and/or returns a function) that will handle exceptions for you. Here is an example:

```
function catchAsync(fn: Function) {
  return function(req: Request, res: Response, next: NextFunction) {
    fn(req, res, next).catch((err: any) => next(err));
  };
}

public getUsers = catchAsync(
  async (req: Request, res: Response, next: NextFunction): Promise<void> => {
    const findAllUsersData: Omit<User, 'password'>[] = await this.userService.findAllUser();
    res.status(200).json({ data: findAllUsersData, message: 'findAll' });
  }
);
```